code/modules/hydroponics/trays/tray_reagents.dm

```diff
@@ -1,126 +1,213 @@
-	toxicity = 8
-	weed_kill_str = 7
-
-// *************************************
-// Nutrient defines for hydroponics
-// *************************************
-
-/obj/item/weapon/reagent_containers/glass/fertilizer
-	name = "fertilizer bottle"
-	desc = "A small glass bottle. Can hold up to 10 units."
-	icon = 'icons/obj/chemical.dmi'
-	icon_state = "bottle16"
-	flags = OPENCONTAINER
-	possible_transfer_amounts = null
-	w_class = 2.0
-
-	var/fertilizer //Reagent contained, if any.
-
-	//Like a shot glass!
-	amount_per_transfer_from_this = 10
-	volume = 10
-
-/obj/item/weapon/reagent_containers/glass/fertilizer/New()
-	..()
-
-	src.pixel_x = rand(-5.0, 5)
-	src.pixel_y = rand(-5.0, 5)
-
-	if(fertilizer)
-		reagents.add_reagent(fertilizer,10)
-
-/obj/item/weapon/reagent_containers/glass/fertilizer/ez
-	name = "bottle of E-Z-Nutrient"
-	icon_state = "bottle16"
-	fertilizer = "eznutrient"
-
-/obj/item/weapon/reagent_containers/glass/fertilizer/l4z
-	name = "bottle of Left 4 Zed"
-	icon_state = "bottle18"
-	fertilizer = "left4zed"
-
-/obj/item/weapon/reagent_containers/glass/fertilizer/rh
-	name = "bottle of Robust Harvest"
-	icon_state = "bottle15"
-	fertilizer = "robustharvest"
+//Process reagents being put into the tray.
+/obj/machinery/portable_atmospherics/hydroponics/proc/process_reagents()
+
+	if(!reagents) return
+
+	if(reagents.total_volume <= 0)
+		return
+
+	for(var/datum/reagent/R in reagents.reagent_list)
+
+		var/reagent_total = reagents.get_reagent_amount(R.id)
+
+		//These are here because they have checks that would clutter up the switch statement cases, and thus get handled after the switch (readability)
+		var/water_value = 0
+		var/health_value = 0
+		var/production_stat_value = 0
+		var/potency_stat_value = 0
+
+
+		switch(R.id)
+		/* EXAMPLE:
+		*
+		*	if("example")
+		*		//Nutrients
+		*		nutrilevel += reagent_total * <nutrient value>
+		*
+		*		//Water
+		*		water_value = <water value>
+		*
+		*		//Toxins
+		*		toxins += reagent_total * <toxin value>
+		*
+		*		//Weeds and Pests
+		*		weedlevel += reagent_total * <weed value>
+		*		pestlevel += reagent_total * <pest value>
+		*
+		*		//Mutagens
+		*		handle_mutagens(reagent_total, <min_value>, <step_value>)
+		*
+		*		//Modifiers
+		*		yield_mod += reagent_total * <yield_mod value>
+		*		mutation_mod += reagent_total * <mutation_mod vlaue>
+		*
+		*		//Health
+		*		health_value = <health value>
+		*
+		*		//Production Stat
+		*		production_stat_value = <production stat value>
+		*
+		*		//Potency Stat
+		*		potency_stat_value = <potency stat value>
+		*
+		*/
+
+			if("adminordrazine")
+				nutrilevel += reagent_total * 1
+				water_value = 1
+				weedlevel += reagent_total * -5
+				pestlevel += reagent_total * -5
+				yield_mod += reagent_total * 1
+				mutation_mod += reagent_total * 1
+				health_value = 1
+			if("ammonia")
+				nutrilevel += reagent_total * 1
+				health_value = 0.5
+			if("atrazine")
+				toxins += reagent_total * 3
+				weedlevel += reagent_total * -8
```

Comment:
I had initially been considering moving all these interactions to a new /datum/reagent proc, but was talked out of it and then realized it would have been worse to maintain that way anyways.

Plus, this is far from the last time i'm touching this code. Just doing some shuffling now before I start considering TG's method of "a proc for each value" on hydro trays.